internal/catch/catch.go

```diff
@@ -0,0 +1,14 @@
+package catch
+
+import (
+	"fmt"
+	"os"
+	"runtime/debug"
+)
+
+func CatchAndLog(err *error, where string) {
```

Comment:
I like the idea, but I don't think `recover` can be moved to a separate function. I added a random `panic` to one of the functions and it didn't work.

Reply:
This works:

```go
package main

import (
	"fmt"
)

func myRecover() {
	if err := recover(); err != nil {
		fmt.Println("reccovered: ", err)
	}
}

func foo() {
	defer myRecover()

	panic("foo")
}

func main() {
	foo()
}
```


I'll see if I can figure out what I did wrong...

Reply:
I've added a test case.